### PR TITLE
Build and Api Versoining

### PR DESCRIPTION
**Pull request by AshantiMutinta**
**Summary of changes**
Changes introduced in this pull request:
- Implements versioning for forest

**Reference issue to close (if applicable)**

Closes 

**Other information and links**

### CODE REVIEW

File: types/src/build_version/mod.rs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+                Command::new("git")
+                    .args(&["rev-parse", "--short", "HEAD", "2>/dev/null"])
+                    .output()
+                    .map(|s| s.stdout)
+                    .unwrap_or_default()
+            });
+        String::from(std::str::from_utf8(&output).unwrap_or_default())
+    };
+    pub static ref BUILD_TYPE: RwLock<BuildType> = RwLock::new(BuildType::BuildDefault);
+    pub static ref RUNNING_NODE_TYPE: RwLock<NodeType> = RwLock::new(NodeType::Full);
```
Comment (austinabell) [inline types/src/build_version/mod.rs:50]: why rwlock? Always same?

File: types/src/build_version/mod.rs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+}
+
+/// represents the current version of the api
+
```
Comment (austinabell) [inline types/src/build_version/mod.rs:54]:
```suggestion
```

File: types/src/build_version/mod.rs [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    }
+}
+
+impl PartialEq for Version {
+    fn eq(&self, other: &Self) -> bool {
+        self.0 & MINOR_MASK == other.0 & MINOR_MASK
+    }
+}
+
+impl ToString for Version {
```
Comment (austinabell) [inline types/src/build_version/mod.rs:126]: why not display?

File: types/src/build_version/mod.rs (hunk 1 above)
Comment (AshantiMutinta) [reply types/src/build_version/mod.rs:50]: Would be injected depending

File: types/src/build_version/mod.rs (hunk 3 above)
Comment (AshantiMutinta) [reply types/src/build_version/mod.rs:125]: matching lotus for concatenation but can change!

**Review by dutterbutter — APPROVED** [review]

File: types/src/build_version/mod.rs (hunk 1 above)
Comment (austinabell) [reply types/src/build_version/mod.rs:50]: The only point I was making here was that there is no need to wrap it in a mutex when it is never overwritten. Doesn't really matter but just no need for the overhead right now

**Review by austinabell — APPROVED** [review]

**Pull request merged**